### PR TITLE
Patch WASM artifacts to run optimized (wormhole enabled) inference

### PR DESCRIPTION
**Pull request by abhi-agg**
This fixes https://github.com/browsermt/bergamot-translator/issues/49

- Patches wasm artifacts to enable wormhole so that optimized inference can happen
- Updated README
- Updated corresponding github workflow

### CODE REVIEW

**Review by jerinphilip — APPROVED** [review]
There are no objections to this pull from our side since this does not appear to break existing functionality in the common area. 

The workflow file seems to include the instructions which should work when the GitHub runners are through. Approved conditioned on the workflow incorporating changes passing.

File: wasm/test_page/start_server.sh [hunk 1]
```diff
@@ -1,8 +1,17 @@
 #!/bin/bash
-
+echo "Start: Copying artifacts in local folder------"
 cp ../../build-wasm/wasm/bergamot-translator-worker.data .
 cp ../../build-wasm/wasm/bergamot-translator-worker.js .
 cp ../../build-wasm/wasm/bergamot-translator-worker.wasm .
 cp ../../build-wasm/wasm/bergamot-translator-worker.worker.js .
+echo "Done----"
+
+echo "Start: Enabling wormhole via APIs that compile and instantiate wasm module-------"
```
Comment (motin) [inline wasm/test_page/start_server.sh:9]: At this point, the artifacts should already be patched, so we should not patch them again (assuming that the user has followed the build steps outlined above)

**Review by motin — APPROVED** [review]
I have left comments suggesting improvements, but I don't want to block merging. Feel free to file my suggestions as issues for future improvements instead.

File: README.md [hunk 2]
```diff
@@ -75,6 +75,12 @@ Bergamot translator provides a unified API for ([Marian NMT](https://marian-nmt.
             emmake make -j
             ```
 
+    3. Enable SIMD Wormhole via Wasm instantiation API in generated artifacts
```
Comment (motin) [inline README.md:78]: With the introduction of this step, please update the phrase further down in this readme that says "As long as you don't update any submodule, just follow steps in 4.ii to recompile."

Comment (abhi-agg) [reply README.md:80]: Done.

File: .github/workflows/macos-custom-marian-wasm.yml [hunk 3]
```diff
@@ -33,6 +33,13 @@ jobs:
         working-directory: build-wasm
         run: emmake make -j2
 
+      - name: Instantiate simd wormhole
+        working-directory: build-wasm
+        run: |
+          sed -i.bak 's/var result = WebAssembly.instantiateStreaming(response, info);/var result = WebAssembly.instantiateStreaming(response, info, {simdWormhole:true});/g' wasm/bergamot-translator-worker.js
```
[the remark this answers is not in the document]
Comment (abhi-agg) [reply .github/workflows/macos-custom-marian-wasm.yml]: Fixed.

File: wasm/test_page/start_server.sh (hunk 1 above)
Comment (abhi-agg) [reply wasm/test_page/start_server.sh]: Agreed.

**Comment by abhi-agg** [comment]
@motin Please review again with the latest change 👍

**Review by motin — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -75,6 +75,12 @@ Bergamot translator provides a unified API for ([Marian NMT](https://marian-nmt.
...
+        ```
+        sed -i.bak 's/var result = WebAssembly.instantiateStreaming(response, info);/var result = WebAssembly.instantiateStreaming(response, info, {simdWormhole:true});/g' wasm/bergamot-translator-worker.js
```
[the remark this answers is not in the document]
Comment (abhi-agg) [reply README.md]: Done.

**Comment by abhi-agg** [comment]
@jerinphilip Could you re-approve?

**Review by jerinphilip — DISMISSED** [review]

**Comment by abhi-agg** [comment]
Just added the missing shell script (this was failing CI wasm builds. Now everything passes.
@jerinphilip @motin Please re-approve. I can also go ahead with the merge as the PR has been approved twice.

**Review by jerinphilip — APPROVED** [review]

**Review by andrenatal — APPROVED** [review]

**Pull request merged**